crates/bevy_ecs/src/component.rs

```diff
@@ -763,6 +764,74 @@ impl Components {
     }
 }
 
+/// A wrapper over a mutable [`Components`] reference that allows for state initialization.
```

Comment:
This should describe how to get one of these on the next line :)

Reply:
Added!